bottles/tests/state/test_events.py

```diff
@@ -0,0 +1,85 @@
+"""EventManager tests"""
```

Comment:
Can we have more comments about how these tests works?
the timing conditions looks scary and not robust enough to me

Reply:
I'm not sure what is the best way to test that events work in this case, with something other than just timing different threads. As per my testing, it seems robust *enough*, but if you have ideas on how to test, I'll be happy to hear them!

Reply:
I think we should test against more specific behavior instead of undistinguished call results.

for example, we should have tests like:
- when we call `wait()` multiple times with same name
  - should only create one `PyEvent` in `EventManager` to make sure they are referencing the same event object
- when we call`done()` or `reset()`
  - the internal flag in the corresponding `PyEvent` should be correctly set
- when we call `done()` or `reset()` with same name in multiple function
  - they should using the same `PyEvent` since the registry in `EventManager` should be singleton